File: samples/generators/angular/src/appgenerated/api/securite/utilisateur/utilisateur.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////


import { HttpClient, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { TypeUtilisateurCode } from "../../../model/securite/utilisateur/references";
import { UtilisateurItem } from "../../../model/securite/utilisateur/utilisateur-item";
import { UtilisateurRead } from "../../../model/securite/utilisateur/utilisateur-read";
import { UtilisateurWrite } from "../../../model/securite/utilisateur/utilisateur-write";
@Injectable({
    providedIn: 'root'
})
export class UtilisateurService {

    constructor(private readonly http: HttpClient) {}

    /**
     * @description Ajoute un utilisateur
     * @param utilisateur Utilisateur à sauvegarder
     * @returns Utilisateur sauvegardé
     */
    addUtilisateur(utilisateur: UtilisateurWrite): Observable<UtilisateurRead> {
        return this.http.post<UtilisateurRead>(`/api/utilisateurs`, utilisateur);
    }

    /**
     * @description Supprime un utilisateur
     * @param utiId Id de l'utilisateur
     */
    deleteUtilisateur(utiId: number): Observable<void> {
        return this.http.delete<void>(`/api/utilisateurs/${utiId}`);
    }

    /**
     * @description Charge le détail d'un utilisateur
     * @param utiId Id de l'utilisateur
     * @returns Le détail de l'utilisateur
     */
    getUtilisateur(utiId: number): Observable<UtilisateurRead> {
        return this.http.get<UtilisateurRead>(`/api/utilisateurs/${utiId}`);
    }

    /**
     * @description Recherche des utilisateurs
     * @param nom Nom de l'utilisateur
     * @param prenom Nom de l'utilisateur
     * @param email Email de l'utilisateur
     * @param dateNaissance Age de l'utilisateur
     * @param adresse Adresse de l'utilisateur
     * @param actif Si l'utilisateur est actif
     * @param profilId Profil de l'utilisateur
     * @param typeUtilisateurCode Type d'utilisateur
     * @returns Utilisateurs matchant les critères
     */
    searchUtilisateur(nom?: string, prenom?: string, email?: string, dateNaissance?: string, adresse?: string, actif?: boolean, profilId?: number, typeUtilisateurCode?: TypeUtilisateurCode, queryParams: any = {}): Observable<UtilisateurItem[]> {
        if(nom) {
            queryParams['nom'] = nom
        }

        if(prenom) {
            queryParams['prenom'] = prenom
        }

        if(email) {
            queryParams['email'] = email
        }

        if(dateNaissance) {
            queryParams['dateNaissance'] = dateNaissance
        }

        if(adresse) {
            queryParams['adresse'] = adresse
        }

        if(actif) {
            queryParams['actif'] = actif
        }

        if(profilId) {
            queryParams['profilId'] = profilId
        }

        if(typeUtilisateurCode) {
            queryParams['typeUtilisateurCode'] = typeUtilisateurCode
        }

        const httpParams = new HttpParams({fromObject : queryParams});
        const httpOptions = { params: httpParams }

        return this.http.get<UtilisateurItem[]>(`/api/utilisateurs`, httpOptions);
    }

    /**
     * @description Sauvegarde un utilisateur
     * @param utiId Id de l'utilisateur
     * @param utilisateur Utilisateur à sauvegarder
     * @returns Utilisateur sauvegardé
     */
    updateUtilisateur(utiId: number, utilisateur: UtilisateurWrite): Observable<UtilisateurRead> {
        return this.http.put<UtilisateurRead>(`/api/utilisateurs/${utiId}`, utilisateur);
    }
}

File: samples/output/jpa/src/main/javagen/topmodel/jpa/sample/demo/entities/securite/utilisateur/Utilisateur_.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package topmodel.jpa.sample.demo.entities.securite.utilisateur;

import jakarta.annotation.Generated;
import jakarta.persistence.StaticMetaModel;

@StaticMetaModel(Utilisateur.class)
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public class Utilisateur_ {

	public static volatile SingularAttribute<Utilisateur, Integer> id;

	public static volatile SingularAttribute<Utilisateur, String> nom;

	public static volatile SingularAttribute<Utilisateur, String> prenom;

	public static volatile SingularAttribute<Utilisateur, String> email;

	public static volatile SingularAttribute<Utilisateur, LocalDate> dateNaissance;

	public static volatile SingularAttribute<Utilisateur, String> adresse;

	public static volatile SingularAttribute<Utilisateur, Boolean> actif;

	public static volatile SingularAttribute<Utilisateur, Profil> profil;

	public static volatile SingularAttribute<Utilisateur, TypeUtilisateur> typeUtilisateur;

	public static volatile SingularAttribute<Utilisateur, LocalDateTime> dateCreation;

	public static volatile SingularAttribute<Utilisateur, LocalDateTime> dateModification;

	public static final String ID = "id";

	public static final String NOM = "nom";

	public static final String PRENOM = "prenom";

	public static final String EMAIL = "email";

	public static final String DATE_NAISSANCE = "dateNaissance";

	public static final String ADRESSE = "adresse";

	public static final String ACTIF = "actif";

	public static final String PROFIL = "profil";

	public static final String TYPE_UTILISATEUR = "typeUtilisateur";

	public static final String DATE_CREATION = "dateCreation";

	public static final String DATE_MODIFICATION = "dateModification";
}
